### PR TITLE
Smart quote handling

### PR DESCRIPTION
**Pull request by graphemecluster**

### CODE REVIEW

File: CantoboardFramework/Keyboard/View/KeyCap.swift [hunk 1]
```diff
@@ -131,9 +131,9 @@ indirect enum KeyCap: Equatable, ExpressibleByStringLiteral {
     var buttonFont: UIFont {
         switch self {
         case .keyboardType(.symbolic), .returnKey(.emergencyCall): return .systemFont(ofSize: 12)
-        case .rime, "^_^", .keyboardType, .returnKey, .space, .character("\t", _, _), .toggleInputMode: return .systemFont(ofSize: 16)
+        case .rime, "^_^", .keyboardType, .returnKey, .space, "\t", .toggleInputMode: return .systemFont(ofSize: 16)
```
Comment (alex-the-man) [inline CantoboardFramework/Keyboard/View/KeyCap.swift]: Would you mind keeping .character("\t", _, _) to be defensive

File: CantoboardFramework/Keyboard/View/KeyCap.swift [hunk 2]
```diff
@@ -234,7 +234,7 @@ indirect enum KeyCap: Equatable, ExpressibleByStringLiteral {
         case .space(.space): return LocalizedStrings.keyTitleSpace
         case .keyboardType(.numeric): return "123"
         case .keyboardType(.symbolic): return "#+="
-        case .keyboardType(.alphabetic): return "ABC"
+        case .keyboardType(.alphabetic): return "ABC" // schema.shortName
```
Comment (alex-the-man) [inline CantoboardFramework/Keyboard/View/KeyCap.swift]: Remove  // schema.shortName

File: CantoboardFramework/Keyboard/View/KeyCap.swift [hunk 3]
```diff
@@ -173,7 +173,7 @@ indirect enum KeyCap: Equatable, ExpressibleByStringLiteral {
     
     var keyCapType: KeyCapType {
         switch self {
-        case .character("\t", _, _): return .system
+        case "\t": return .system
```
Comment (alex-the-man) [inline CantoboardFramework/Keyboard/View/KeyCap.swift:155]: Revert this change

File: CantoboardFramework/Keyboard/View/KeyCap.swift [hunk 4]
```diff
@@ -457,15 +469,15 @@ indirect enum KeyCap: Equatable, ExpressibleByStringLiteral {
     var keyCapCharacter: String? {
         switch self {
         case .character(let c, _, _), .cangjie(let c, _): return c.lowercased()
+        case .currency: return "$"
```
[the remark this answers is not in the document]
Comment (alex-the-man) [reply CantoboardFramework/Keyboard/View/KeyCap.swift:446]: Do you mind if you revert this change?

File: CantoboardFramework/Keyboard/View/KeyCap.swift [hunk 5]
```diff
@@ -457,15 +469,15 @@ indirect enum KeyCap: Equatable, ExpressibleByStringLiteral {
...
         default: return nil
         }
     }
     
     var isCharacter: Bool {
-        switch self {
-        case .character, .cangjie: return true
-        default: return false
-        }
+        keyCapCharacter != nil
```
[the remark this answers is not in the document]
Comment (alex-the-man) [reply CantoboardFramework/Keyboard/View/KeyCap.swift:454]: Do you mind if you revert this change?

File: CantoboardFramework/Keyboard/View/Keyboard/KeyboardViewLayout.swift [hunk 6]
```diff
@@ -29,10 +29,7 @@ protocol KeyboardViewLayout {
 
 extension KeyboardViewLayout {
     static func isSwipeDownKeyShiftMorphing(keyCap: KeyCap) -> Bool {
-        switch keyCap {
-        case .character(let c, _, _), .cangjie(let c, _): return !(c.first?.isLetter ?? false)
-        default: return true
-        }
+        return !(keyCap.keyCapCharacter?.first?.isEnglishLetter ?? false)
```
[the remark this answers is not in the document]
Comment (alex-the-man) [reply CantoboardFramework/Keyboard/View/Keyboard/KeyboardViewLayout.swift:32]: Mind if you keep it as is?

File: CantoboardFramework/Keyboard/View/Keyboard/PadFull5RowsKeyboardViewLayout.swift [hunk 7]
```diff
@@ -113,10 +113,13 @@ class PadFull5RowsKeyboardViewLayout : KeyboardViewLayout {
                 } else {
                     width = padFull5RowsLayoutConstants.largeSystemKeyWidth
                 }
-            case .character, .cangjie, .currency: width = inputKeyWidth
             default:
-                width = 0
-                numFlexibleWidthKeys += 1
+                if (keyCap.isCharacter) {
```
Comment (alex-the-man) [inline CantoboardFramework/Keyboard/View/Keyboard/PadFull5RowsKeyboardViewLayout.swift:117]: Mind if you revert this change too?

**Comment by alex-the-man** [comment]
Will merge after reverting the risky refactoring. Thank you.

**Comment by graphemecluster** [comment]
Completed. Thank you.

**Comment by alex-the-man** [comment]
Thank you so much for adding this

**Comment by graphemecluster** [comment]
Sorry to miss 2 lines while merging. Please merge #53 accordingly. Thanks again.
